### PR TITLE
feat: Confirmation redesign - Remove the scheme ("https://","http://") from the InfoRowUrl value

### LINKED ISSUE #23968: Confirmation redesign - Remove the scheme ("https://","http:") from the InfoRowUrl value
### What is this about?

Remove the scheme ("https://","http://") from the InfoRowUrl value. If the URL displayed is "HTTP" it is shown as a label to the left of the text value. There is no need to display the scheme of the URL text value.

### Scenario

_No response_

### Design

Figma: https://www.figma.com/file/7kYok9B46ohR6h47cksRAz/Confirmation-redesign-V1?type=design&node-id=1735-8556&mode=design&t=6PRcdkYIuz1RF6ek-0
Storybook Prod: https://metamask.github.io/metamask-storybook/?path=/docs/components-app-confirm-inforowurl--docs

Screenshot 2024-04-11 at 3 02 25 PM

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by segun**
## **Description**

Remove the scheme ("https://","http://") from the InfoRowUrl value. If the URL displayed is "HTTP" it is shown as a label to the left of the text value. There is no need to display the scheme of the URL text value.

Figma: https://www.figma.com/file/7kYok9B46ohR6h47cksRAz/Confirmation-redesign-V1?type=design&node-id=1735-8556&mode=design&t=6PRcdkYIuz1RF6ek-0
Storybook Prod: https://metamask.github.io/metamask-storybook/?path=/docs/components-app-confirm-inforowurl--docs

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24105?quickstart=1)

## **Related issues**

Fixes: #23968 

## **Manual testing steps**

1. Enable conf redesign in .metamaskrc (ENABLE_CONFIRMATION_REDESIGN=1)
2. Run metamask
3. Open an http url
4. Type the code below in the console
5. You should see the HTTP label
6. Open testdapp and click on Personal Sign
7. You should not see the HTTP label

```
ethereum.request({
  method: 'wallet_requestPermissions',
  params: [{ eth_accounts: {} }],
});


const msg = "Hello World".toString('hex');

const from = [the connected account]

ethereum.request({
  method: 'personal_sign',
  params: [msg, from, 'Example password'],
});
```

### **Before**

Screenshot 2024-04-18 at 11 08 12

### **After**

Screenshot 2024-04-18 at 11 14 27

Screenshot 2024-04-18 at 11 15 00

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by cryptotavares — APPROVED** [review]

**Comment by jpuri** [comment]
@segun : changes look good to me, it will be nice to run these through snaps team also once as they also use these components I guess.

**Review by jpuri — APPROVED** [review]

File: ui/components/app/confirm/info/row/url.tsx [hunk 1]
```diff
@@ -32,6 +32,8 @@ export const ConfirmInfoRowUrl = ({ url }: ConfirmInfoRowUrlProps) => {
 
   const isHTTP = urlObject?.protocol === 'http:';
 
+  const urlWithoutProtocol = url?.replace(/https?:\/\//u, '');
```
Comment (digiwand) [inline ui/components/app/confirm/info/row/url.tsx:35]: `http` is also a valid SIWE schema. We will need to add this to the regex

**Review by digiwand — CHANGES_REQUESTED** [review]
-

**Review by digiwand — APPROVED** [review]
Nice work! 👍🏼

**Pull request merged**